### PR TITLE
 add github pre-commit (checks for whitespace, newline etc)

### PR DESCRIPTION
**Pull request by ideepika**
github precommit can be used to scrape trailing spaces, new endline and
other common linting from the to be committed patch, for which reviewer
right now manually has to review

How it looks: 
```
❯ git commit --amend
Check Yaml...........................................(no files to check)Skipped
Fix End of Files.........................................................Passed
Trim Trailing Whitespace.................................................Passed
black................................................(no files to check)Skipped
```

Signed-off-by: Deepika Upadhyay <dupadhya@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: install-deps.sh [hunk 1]
```diff
@@ -465,6 +465,10 @@ function preload_wheels_for_tox() {
     popd > /dev/null
 }
 
+# install git precommit hooks https://pre-commit.com/
+curl https://pre-commit.com/install-local.py | python -
+pre-commit install > /dev/null
```
Comment (tchaikov) [inline install-deps.sh:470]: i'd avoid putting this kind of opinionated setting in `install-deps.sh`, which is supposed to ready the build dependencies and only the build dependencies.

Comment (ideepika) [reply install-deps.sh]: i agree, unable to find any place where we can enforce this, added in a separate script

**Comment by batrick** [comment]
I'm not sure what dark magic is involved with this but it looks like you're wanting a pre-commit hook for git that checks whitespace? Why wouldn't we use a jenkins job?

There's a reason git makes it difficult to setup ecosystem wide commit scripts: it's an obvious security vulnerability. I don't think we can use this as a mechanism realistically. It needs to be a jenkins job.

**Comment by ideepika** [comment]
> I'm not sure what dark magic is involved with this but it looks like you're wanting a pre-commit hook for git that checks whitespace? Why wouldn't we use a jenkins job?
> 
> There's a reason git makes it difficult to setup ecosystem wide commit scripts: it's an obvious security vulnerability. I don't think we can use this as a mechanism realistically. It needs to be a jenkins job.

thanks @batrick ! I had my doubts as well, wanted to see what others think. Jenkins job seems like a good alternative, also see we already run some checks, will add to them if needed

**Pull request closed without merging**